class3.js

```diff
@@ -0,0 +1,136 @@
+// Exercises to practices loops! aka rabbit holes 🤣
+
+// Exercise 1)
+// Get the sum of all the elements of an array of numbers
+
+let exampleArray1 = [3, 5, 22, 5, 7, 2, 45, 75, 89, 21, 2];
+
+function sumOfAllElements(array) {
+  let total = 0;
+  for (let index = 0; index < array.length; index++) {
+    const element = array[index];
+    total = total + element;
+  }
+  return total;
+}
+
+console.log(
+  "Exercise 1 result (sumOfAllElements) is:",
+  sumOfAllElements(exampleArray1),
+  "it should be",
+  276
+);
+
+// Exercise 2)
+// Using a for loop output the elements in reverse order
+let arr = [43, "what", 9, true, "cannot", false, "be", 3.5, true];
+
+function reverseOrder(array) {
+  let newArray = [];
+  for (let index = array.length - 1; index > 0; index--) {
+    const element = array[index];
+    newArray.push(element);
+  }
+  return newArray;
+}
+
+console.log(
+  "Exercise 2 result (reverseOrder) is:",
+  reverseOrder(arr),
+  "it should be",
+  [true, 3.5, "be", false, "cannot", true, 9, "what", 43]
+);
+// Exercise 3)
+// Giving an array of numbers, return a new array which contains only the even numbers
+let numbers = [78, 5, 22, 5, 7, 2, 45, 75, 16, 21, 2];
+// you may want to see this to learn how to check even numbers -> https://www.programiz.com/javascript/examples/even-odd#:~:text=In%20the%20above%20program%2C%20number,Hence%2C%20the%20number%20is%20odd.
+function getEvenNumbers(array) {
+  let newArray = [];
+  for (let index = 0; index < array.length; index++) {
+    const element = array[index];
+    if (element % 2 == 0) {
+      newArray.push(element);
+    }
+  }
+  return newArray;
+}
+
+console.log(
+  "Exercise 3 result (getEvenNumbers) is:",
+  getEvenNumbers(numbers),
+  "it shold be",
+  [78, 22, 2, 16, 2]
+);
+
+// Exercise 4)
+// Given a string change the every second letter to an uppercase ‘Z’. Assume
+// there are no spaces.
+let str1 = "javascript";
+
+function addZToString(myString) {
+  let myNewString = [];
+  myString.split();
```

Comment:
This `myString.split()` has no effect in your function. According to the bible (MDN):

The `split()` method divides a String into an ordered list of substrings, puts these substrings **into an array**, and **returns the array**.

You can see that it will return a new array, but you are not assigning that to any variable or doing anything with this new array that `myString.split()` returns. You could remove it from your function 👍